streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -588,12 +588,16 @@ class StreamingContext private[streaming] (
     state match {
       case INITIALIZED =>
         startSite.set(DStream.getCreationSite())
-        sparkContext.setCallSite(startSite.get)
         StreamingContext.ACTIVATION_LOCK.synchronized {
           StreamingContext.assertNoOtherContextIsActive()
           try {
             validate()
-            scheduler.start()
+            ThreadUtils.runInNewThread("streaming-start") {
```

Comment:
this block definitely requires a paragraph comment. Someone new to the code will have no idea why we need to do these things in a new thread.